Sources/ProjectSpec/SpecFile.swift

```diff
@@ -26,22 +27,50 @@ public struct SpecFile {
         static let defaultEnable = true
 
         init?(any: Any) {
-            if let string = any as? String {
-                path = Path(string)
-                relativePaths = Include.defaultRelativePaths
-                enable = Include.defaultEnable
+            if let path = any as? String {
+                self.init(
+                    path: Path(path),
+                    relativePaths: Include.defaultRelativePaths,
+                    enable: Include.defaultEnable
+                )
             } else if let dictionary = any as? JSONDictionary, let path = dictionary["path"] as? String {
-                self.path = Path(path)
-                relativePaths = Self.resolveBoolean(dictionary, key: "relativePaths") ?? Include.defaultRelativePaths
-                enable = Self.resolveBoolean(dictionary, key: "enable") ?? Include.defaultEnable
+                self.init(
+                    path: Path(path),
+                    dictionary: dictionary
+                )
             } else {
                 return nil
             }
         }
+        
+        private init(path: Path, relativePaths: Bool, enable: Bool) {
+            self.path = path
+            self.relativePaths = relativePaths
+            self.enable = enable
+        }
+        
+        private init?(path: Path, dictionary: JSONDictionary) {
+            self.path = path
+            relativePaths = Self.resolveBoolean(dictionary, key: "relativePaths") ?? Include.defaultRelativePaths
+            enable = Self.resolveBoolean(dictionary, key: "enable") ?? Include.defaultEnable
+        }
+        
+        private static func includes(from array: [Any], basePath: Path) -> [Include] {
+            array.flatMap { entry -> [Include] in
+                if let string = entry as? String, let include = Include(any: string) {
```

Comment:
If I'm reading this right only includes defined as a dictionary will be glob enabled, whereas plain strings won't?

Reply:
yeah, I wanted to keep the changes as isolated to my use case as possible.
But I can have a look at extending it if you want to

Reply:
started working on this, but still ironing out some stuff

Reply:
implemented